scrooge-sbt-plugin/src/main/scala/com/twitter/ScroogeSBT.scala

```diff
@@ -179,8 +188,9 @@ object ScroogeSBT extends AutoPlugin {
       streams,
       scroogeThriftSources,
       scroogeThriftOutputFolder,
-      scroogeThriftIncludes
-    ) map { (out, sources, outputDir, inc) =>
+      scroogeThriftIncludes,
+      scroogeGenerated
```

Comment:
The current check is simple - check all the *.scala files in the output directory (or .java - support was added in cd5e6c0b6f3e0).

You could figure out the output files by parsing the input and figuring out what structs and services would be generated. This is more correct but too complex. I don't know if there is a good solution for this.